contrib/platform/src/com/sun/jna/platform/win32/Winioctl.java

```diff
@@ -72,4 +95,320 @@ public STORAGE_DEVICE_NUMBER(Pointer memory) {
+         */
+        public WinDef.DWORD rawPropertiesLength;
+
+        /**
+         * Contains an array of length one that serves as a place holder for the first byte of the bus specific property
+         * data.
+         */
+        public byte[] rawDeviceProperties = new byte[1];
+
+        public static class ByReference extends STORAGE_DEVICE_DESCRIPTOR implements Structure.ByReference {
+            public ByReference() {
+            }
+
+            public ByReference(Pointer memory) {
+                super(memory);
+            }
+        }
+
+        public STORAGE_DEVICE_DESCRIPTOR() {
+        }
+
+        public STORAGE_DEVICE_DESCRIPTOR(Pointer memory) {
+            super(memory);
+            read();
+
+            // Allocate the buffer for the raw properties, and re-read
+            rawDeviceProperties = new byte[rawPropertiesLength.intValue()];
+            read();
+        }
+
+        @Override
+        protected List<String> getFieldOrder() {
+            return FIELDS;
+        }
+    }
+
+    /**
+     * Describes the extended geometry of disk devices and media.
+     */
+    class DISK_GEOMETRY_EX extends Structure {
+        public static class ByReference extends DISK_GEOMETRY_EX implements Structure.ByReference {
+            public ByReference() {
+            }
+
+            public ByReference(Pointer memory) {
+                super(memory);
+            }
+        }
+
+        public static final List<String> FIELDS = createFieldsOrder("geometry", "diskSize", "data");
+
+        public DISK_GEOMETRY_EX() {
+        }
+
+        public DISK_GEOMETRY_EX(Pointer memory) {
+            super(memory);
+            read();
+        }
+
+        /**
+         * A DISK_GEOMETRY structure.
+         */
+        public DISK_GEOMETRY geometry;
+
+        /**
+         * The disk size, in bytes.
+         */
+        public WinNT.LARGE_INTEGER diskSize;
+
+        /**
+         * Any additional data.
+         */
+        public WinDef.BYTE[] data = new WinDef.BYTE[0x200];
+
+        @Override
+        public String toString() {
+            return String.format(Locale.ROOT, "disk-geometry-ex[cylinders:%d tracksPerCylinder:%d sectorsPerTrack:%d" +
+                    " bytesPerSector:%d] mediaType:%d diskSize:%d",
+                    geometry.cylinders.longValue(), geometry.tracksPerCylinder.longValue(),
+                    geometry.sectorsPerTrack.longValue(), geometry.bytesPerSector.longValue(),
+                    geometry.mediaType.longValue(), diskSize.getValue());
+        }
+
+        @Override
+        protected List<String> getFieldOrder() {
+            return FIELDS;
+        }
+    }
+
+    /**
+     * Describes the geometry of disk devices and media.
+     */
+    class DISK_GEOMETRY extends Structure {
+        public static class ByReference extends DISK_GEOMETRY implements Structure.ByReference {
+            public ByReference() {
+            }
+
```

Comment:
Don't leave this empty - add call to (implicit) _super()_ as this enables setting a breakpoint in the constructor